### PR TITLE
Add segment tracking to ecosystem page

### PR DESCRIPTION
**Pull request by frankharkins**
## Changes

Adds segment tracking to qiskit.org/ecosystem, specifically
- filters
- search
- sorting

### CODE REVIEW

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you Frank! I've 2 questions and 1 request for change

File: pages/ecosystem.vue [hunk 1]
```diff
@@ -258,6 +263,7 @@ const selectedSortingOption = ref<SortingOption>("default");
 
 function updateSelectedSortingOption(sortingOption: SortingOption) {
   selectedSortingOption.value = sortingOption;
+  trackClickEvent(sortingOption, "ecosystem-search");
```
[the remark this answers is not in the document]
Comment (frankharkins) [reply pages/ecosystem.vue]: Oops, thanks!

File: pages/ecosystem.vue [hunk 2]
```diff
@@ -201,6 +203,7 @@ const tierFiltersAsString = computed(() => tierFilters.value.join(","));
 
 function updateCategoryFilter(filterValue: string, isChecked: boolean) {
   if (isChecked) {
+    trackClickEvent(filterValue, "ecosystem-filters");
```
[the remark this answers is not in the document]
Comment (frankharkins) [reply pages/ecosystem.vue:206]: Yep! Just want to see if someone uses the box, but don't want to count it twice if they uncheck it

File: pages/ecosystem.vue [hunk 3]
```diff
@@ -222,6 +225,7 @@ function isCategoryFilterChecked(filterValue: string): boolean {
 
 function updateTierFilter(filterValue: string, isChecked: boolean) {
   if (isChecked) {
+    trackClickEvent(filterValue, "ecosystem-filters");
```
[the remark this answers is not in the document]
Comment (frankharkins) [reply pages/ecosystem.vue:228]: Yep, same answer :)

**Review by y4izus — APPROVED** [review]
LGTM!

**Pull request merged**